packages/devtools-reps/src/object-inspector/tests/component/state.js

```diff
@@ -0,0 +1,158 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const { mount } = require("enzyme");
+const React = require("react");
+const { createFactory } = React;
+const ObjectInspector = createFactory(require("../../index"));
+const repsPath = "../../../reps";
+const { MODE } = require(`${repsPath}/constants`);
+const ObjectClient = require("../__mocks__/object-client");
+const gripRepStubs = require(`${repsPath}/stubs/grip`);
+const gripPropertiesStubs = require("../../stubs/grip");
+
+function generateDefaults(overrides) {
+  return Object.assign({
+    autoExpandDepth: 0,
+    roots: [{
+      path: "root-1",
+      contents: {
+        value: gripRepStubs.get("testMoreThanMaxProps")
+      }
+    }, {
+      path: "root-2",
+      contents: {
+        value: gripRepStubs.get("testProxy")
+      }
+    }],
+    createObjectClient: grip => ObjectClient(grip),
+    mode: MODE.LONG,
+  }, overrides);
+}
+
+describe("ObjectInspector - state", () => {
+  it("has the expected expandedPaths state", () => {
+    const wrapper = mount(ObjectInspector(generateDefaults({
+      loadedProperties: new Map([
+        ["root-1", gripPropertiesStubs.get("proto-properties-symbols")]
+      ])
+    })));
+
+    let nodes = wrapper.find(".node");
+
+    // Clicking on the root node adds it path to "expandedPaths".
+    const root1 = nodes.at(0);
+    const root2 = nodes.at(1);
+
+    root1.simulate("click");
+    expect(wrapper.state("expandedPaths").has("root-1")).toBeTruthy();
+    expect(wrapper.state("expandedPaths").has("root-2")).toBeFalsy();
+
+    // Clicking on the root node removes it path from "expandedPaths".
+    root1.simulate("click");
+    expect(wrapper.state("expandedPaths").has("root-1")).toBeFalsy();
+    expect(wrapper.state("expandedPaths").has("root-2")).toBeFalsy();
+
+    root2.simulate("click");
+    expect(wrapper.state("expandedPaths").has("root-1")).toBeFalsy();
+    expect(wrapper.state("expandedPaths").has("root-2")).toBeTruthy();
+
+    root1.simulate("click");
+    expect(wrapper.state("expandedPaths").has("root-1")).toBeTruthy();
+    expect(wrapper.state("expandedPaths").has("root-2")).toBeTruthy();
```

Comment:
it might be nice to have many snapshots w/ formatted trees.

click 1 => formatted tree
click 2 => formatted tree
click 3 => formatted tree
click 4 => formatted tree

snapshot TV like tests
https://www.youtube.com/watch?v=HAuXJVI_bUs